jupyterhub/templates/hub/deployment.yaml

```diff
@@ -230,3 +230,6 @@ spec:
               path: {{ .Values.hub.baseUrl | trimSuffix "/" }}/hub/health
               port: http
           {{- end }}
+      {{- if .Values.hub.extraPodSpec }}
+      {{ toYaml .Values.hub.extraPodSpec | nindent 6 }}
+      {{- end }}
```

Comment:
```suggestion
      {{- with .Values.hub.extraPodSpec }}
      {{- . | toYaml | trimSuffix "\n" | nindent 6 }}
      {{- end }}
```

Reply:
Note that `with` is conditional, if its a falsy value - it will not evaluate anything inside its block. `trimSuffix "\n"` is added because `toYaml` has a quirk of always adding a `\n` in the end which makes the rendered YAML look a bit ugly.